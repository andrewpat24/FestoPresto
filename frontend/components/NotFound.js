import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
    <h1>404 - <Link to="/">Go Home</Link></h1>
)

export default NotFound; 