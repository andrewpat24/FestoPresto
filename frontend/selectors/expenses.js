const getVisibleExpenses = (
  expenses,
  /* filters:*/ { text, sortBy, startDate, endDate }
) => {
  return expenses
    .filter(expense => {
      const startDateMatch = !startDate || expense.createdAt >= startDate;
      const endDateMatch = !endDate || expense.createdAt <= endDate;
      const textMatch = expense.description.toLowerCase().includes(text);
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createdAt < b.createdAt ? 1 : -1;
      } else if (sortBy === "amount") {
        return a.amount < b.amount ? 1 : -1;
      }
    });
};

export default getVisibleExpenses;
