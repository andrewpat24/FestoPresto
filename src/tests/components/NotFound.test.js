import React from "react";
import { shallow } from "enzyme";
import NotFound from "../../components/NotFound";
import expenses from "../fixtures/expenses";

test("Should render NotFound component correctly", () => {
  const wrapper = shallow(<NotFound />);
  expect(wrapper).toMatchSnapshot();
});

// test("Should render ExpenseListItem with expense element", () => {
//     const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
//     expect(wrapper).toMatchSnapshot();
//   });
