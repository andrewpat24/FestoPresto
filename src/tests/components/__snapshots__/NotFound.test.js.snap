// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NotFound component correctly 1`] = `
<h1>
  404 - 
  <Link
    to="/"
  >
    Go Home
  </Link>
</h1>
`;
